### PR TITLE
DM-26550: Add support for ObservationInfo.observation_reason

### PR DESCRIPTION
**Pull request by timj**
This allows TESTTYPE handling to be unified.

Requires lsst/astro_metadata_translator#40

### CODE REVIEW

File: config/latiss/ingest.py [hunk 1]
```diff
@@ -24,8 +24,7 @@
 
 config.parse.retarget(LatissParseTask)
 
-del config.parse.translation['testType']
-
+del config.parse.translators['testType']
```
Comment (RobertLuptonTheGood) [inline config/latiss/ingest.py:27]: Do we need this white-space only change?

**Review by RobertLuptonTheGood — APPROVED** [review]
I don't think that the test `observation_reason`s are very realistic, but as long as no-one cargo cults them that's OK.

File: config/latiss/ingest.py (hunk 1 above)
Comment (timj) [reply config/latiss/ingest.py:27]: It's not a whitespace only change.

**Comment by timj** [comment]
I don't think I can do anything other than believe the value of the TESTTYPE header can I? Garbage in and garbage out. If you think that TESTTYPE=bias could be translated to something else then I'm happy to tweak the translation here.

**Comment by RobertLuptonTheGood** [comment]
Tim explained that the test code was using real `TESTTYPE` values, so my comment is wrong.   Good to merge

**Pull request merged**